nautobot/ui/src/components/ObjectTable.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useState, useMemo, useCallback } from "react";
+import {
+    TableRenderer,
+    useTableRenderer,
+    createColumnHelper,
+    Button,
+    EditIcon,
+} from "@nautobot/nautobot-ui";
+
+import ObjectTableItem from "@components/ObjectTableItem";
+
+function getTableItemLink(idx, obj) {
+    if (idx === 0) {
+        // TODO: ui-schema should be providing the name of the field used to describe the object (name)
+        return window.location.pathname + obj.id + "/";
+    }
+    if (typeof obj !== "object" || !obj || !obj.url) {
+        return null;
+    }
+    // Remove domain + /api prefix
+    return obj.url.replace(window.location.origin + "/api", "");
+}
+
+function createTableColumn(tableHeaders, columnHelper) {
+    return tableHeaders.map(({ key, title }, idx) =>
+        columnHelper.accessor(key, {
+            cell: (props) => {
+                // Get the column data from the object
+                // e.g from {"status": {"display": "Active"}, "id": ....} get => {"display": "Active"}
+                const column_data =
+                    idx === 0
+                        ? props.row.original
+                        : props.row.original[props.column.id];
+                return (
+                    <ObjectTableItem
+                        name={key}
+                        obj={props.getValue()}
+                        url={getTableItemLink(idx, column_data)}
+                    />
```

Comment:
Is the comment inaccurate here? I don't see how `column_data = {"display": "Active"}` then leads to a useful `getTableItemLink(idx, {"display": "Active"})` call when it seems like `getTableItemLink()` expects an object with an `id` or `url` key.